### ISSUE TITLE
[BUG][benchmarks]: benchmark_grpo_op and benchmark_sampling produce incorrect measurements due to flawed VRAM baseline and missing TRL comparison

### ISSUE THREAD

**Issue by Flink-ddd** [Flink-ddd/RL-Engine#78]
## Bug Description

Two benchmark scripts contain measurement issues that produce misleading results:

### 1. benchmarks/benchmark_grpo_op.py incorrect VRAM baseline

**Problem:** The script allocates the input logits tensor [G, SeqLen, Vocab] before calling reset_peak_memory_stats(). This means both native and RL-Kernel measurements include the input tensor itself in their peak VRAM readings, causing RL-Kernel to appear *higher* than native in some cases, the opposite of the true result.

**Root cause:**
```python
# Current (wrong): logits allocated before baseline is recorded
logits = torch.randn(g, seq_len, vocab_size, ...)
torch.cuda.reset_peak_memory_stats()  # too late — logits already allocated
```

**Fix:** Record baseline *after* input allocation, measure only the extra overhead above input tensors during the compute call.

---

### 2. benchmarks/benchmark_sampling.py missing TRL baseline

**Problem:** The script only compares Native PyTorch vs RL-Kernel. TRL (selective_log_softmax) is the primary baseline users care about, and its absence makes the benchmark incomplete for real-world comparisons.

---

## Fix

- benchmark_grpo_op.py: rewrite VRAM measurement to use pre/post baseline delta, isolating extra overhead from input tensor size
- benchmark_grpo_op.py: add TRL selective_log_softmax as a third comparison column inlined to avoid transformers version conflicts
- benchmark_sampling.py: fix single-measurement timing replace time.perf_counter` wall-clock with CUDA events for accurate GPU timing

## Verified Results After Fix (NVIDIA A100-SXM4-80GB, SeqLen=512, Vocab=128256)

| G | Native VRAM | TRL VRAM | RL-Kernel VRAM | Native ms | TRL ms | RL-Kernel ms | Speedup vs TRL |
|---|---|---|---|---|---|---|---|
| 8 | 3.91 GB | 0.25 GB | 0.98 GB | 7.81 ms | 2.76 ms | 2.53 ms | 1.09x |
| 16 | 7.83 GB | 0.25 GB | 1.96 GB | 15.08 ms | 5.39 ms | 5.03 ms | 1.07x |
| 32 | 15.66 GB | 0.25 GB | 3.91 GB | 30.13 ms | 10.77 ms | 10.02 ms | 1.08x |
| 64 | 31.31 GB | 0.25 GB | 7.83 GB | 60.17 ms | 21.54 ms | 20.00 ms | 1.08x |
| 128 | 62.62 GB | 0.25 GB | 15.66 GB | 119.79 ms | 43.08 ms | 39.16 ms | 1.10x |
| 256 | OOM | 0.25 GB | 31.31 GB | OOM | 87.91 ms | 78.37 ms | 1.12x |

## Files Changed
- benchmarks/benchmark_grpo_op.py
- benchmarks/benchmark_sampling.py
